### PR TITLE
Refactor code to use the common messaging component

### LINKED ISSUE #66: Refactor existing alerts to new messaging component
New messaging component is described [./doc/components-libraries.md](doc/components-libraries.md). Some additional info is described in https://github.com/kbss-cvut/record-manager-ui/pull/65, where it is explained that it should solve the following issues:

- 1) Cannot see newest message:
image

- 2) I would like to dismiss notification of records deleted (maybe not part of this issue):
image

A/C:
- all alerts are migrated to the new component
- two issues described above are solved

### PR DESCRIPTION
**Pull request by ledsoft**
Resolves #66.

The alert messages on public routes (login, password reset, password token) are kept as is, because the common messaging component is available only in the MainView, which requires authentication.

### CODE REVIEW

File: tests/__tests__/actions/InstitutionActions.spec.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ describe('Institution asynchronize actions', function () {
         store.dispatch(createInstitution(institution));
 
         setTimeout(() => {
-            expect(store.getActions()).toEqual(expectedActions);
+            expect(store.getActions().slice(0, 3)).toEqual(expectedActions);
```
[the remark this answers is not in the document]
Comment (blcham) [reply tests/__tests__/actions/InstitutionActions.spec.js:163]: This way, it seams to me easier to understand and reuse without making a mistake.

[the remark this answers is not in the document]
Comment (blcham) [reply tests/__tests__/actions/InstitutionActions.spec.js:163]: See also other occurrences of this pattern within this file.

**Review by blcham — APPROVED** [review]
LGTM, nice cleanup of the code! Can be merged but,
consider two things to fix:
- slice(0, ?) as mentioned in comment
- I believe you removed all tests testing for alert messages within components. Isn't it useful to know for a component that it triggers alert messages correctly? Or it would be difficult to test now after refactoring ?

**Comment by ledsoft** [comment]
- Fixed action tests - removed the `slice` calls and instead mock `Date.now()` to provide the same timestamp so that publish message actions can be checked as well
- Yes, I did remove all the alert message tests from components, as the components no longer show any alerts. Messaging is now done via actions and publish message actions are now checked in tests as well (see the previous point)

**Pull request merged**